wordpress-mobile/WordPress-Android
# Remove My Site Dashboard default tab a/b experiment

Closes #16202 

This PR disables and removes the My Site Dashboard - Default tab experiment
See parent issue for all the details.

Moving forward the default tab will be set as follows:
-For Jetpack App: Home
-For WordPress App: Menu

To support the different default values for the "initial screen" (in App Settings), `initial_screen_entry_value_default_key` was added to Jetpack `strings.xml`. The actual “initial screen” app setting is NOT set until the user navigates to Me → App Setting view and manually changes. The default value is used until then.

### Tests
**_Please execute tests in sequence_**

**Jetpack: Test A: Initial tab on signup**
- Do a fresh install of the Jetpack App
- Sign up with a new account
- After signup, follow the "Create new site" flow
- After you site is created, select No thanks or no or dismiss to the prompt pop-ups
- ✅ Verify the selected tab is `Home`

**Jetpack: Test B: Initial Screen**
- Navigate to `My Site` -> `Me` -> `App Settings` 
- Tap Privacy Setting -> then enable Collection Information
- Navigate back to App Settings
- ✅ Verify `Initial Screen` is set to `Home`
- Tap `Initial Screen` and select `Menu` to change the initial screen
- Navigate back to My Site
- ✅ Verify the selected tab is now `Menu`

**Jetpack: Test C: tracking**
- Navigate to `My Site` -> `Me`  -> Help and Support -> Application Log  and verify the following:
- ✅ 🔵 Tracked: app_settings_initial_screen_changed, Properties: {"selected":"menu"} 
- ✅ Verify "default_tab_experiment" is not being collected in ANY tracked events

**Jetpack: Test D: Initial Screen change is respected on app reopen**
- Close the App (not background)
- Reopen the app
- ✅ Verify the selected tab is the `Menu` tab
- Log out of the app

**Jetpack: Test E: Variant assigned upon login is reflected in the selected tab of My Site view**
- Ensure you have logged out of the app
- Log in
- ✅ Verify the selected tab matches the variant just assigned

____________________________________________________
**WordPress: Test A: Initial tab on signup**
- Do a fresh install of the WordPress App
- Sign up with a new account
- After signup, follow the "Create new site" flow
- After you site is created, select No thanks or no or dismiss to any prompt pop-ups
- ✅ Verify the selected tab is `Menu`

**WordPress: Test B: Initial Screen**
- Navigate to `My Site` -> `Me` -> `App Settings` 
- Tap Privacy Setting -> then enable Collection Information
- Navigate back to App Settings
- ✅ Verify `Initial Screen` is set to `Menu`
- Tap `Initial Screen` and select `Home` to change the initial screen
- Navigate back to My Site
- ✅ Verify the selected tab is now `Home`

**WordPress: Test C: tracking**
- Navigate to `My Site` -> `Me`  -> Help and Support -> Application Log  and verify the following:
- ✅ 🔵 Tracked: app_settings_initial_screen_changed, Properties: {"selected":"dashboard"} 
- ✅ Verify "default_tab_experiment" is not being collected in ANY tracked events

**WordPress: Test D: Initial Screen change is respected on app reopen**
- Close the App (not background)
- Reopen the app
- ✅ Verify the selected tab is the `Home` tab
- Log out of the app

**WordPress: Test E: Variant assigned upon login is reflected in the selected tab of My Site view**
- Ensure you have logged out of the app
- Log in
- ✅ Verify the selected tab matches the variant just assigned
- 
Extra credit:  If you are testing on an emulator and want to check the preferences values as you move along, then follow these instructions
    - Open Device File Explorer
    - WordPress: Navigate to data -> data -> org.wordpress.android.prealpha -> shared_prefs
    - Double click on `org.wordpress.android.prealpha_preferences.xml` to open the file
    - Jetpack: Navigate to data -> data -> com.jetpack.android.prealpha -> shared_prefs
    - Double click on `com.jetpack.android.prealpha_preferences.xml` to open the file

## Regression Notes
1. Potential unintended areas of impact
The tab selected in My Site view does not reflect initial screen 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Removed experiment tests

PR submission checklist:
